### PR TITLE
C6: Basic LP_UART driver

### PR DESCRIPTION
**Pull request by JurajSadel**
## Thank you!

Thank you for your contribution.
Please make sure that your submission includes the following:

### Must

- [x] The code compiles without `errors` or `warnings`.
- [x] All examples work.
- [x] `cargo fmt` was run.
- [ ] Your changes were added to the `CHANGELOG.md` in the proper section.
- [x] You updated existing examples or added examples (if applicable).
- [x] Added examples are checked in CI

### Nice to have

- [x] You add a description of your work to this PR.
- [ ] You added proper docs for your newly added features and code.

This adds a basic LP_UART driver for C6.
Currently, it's not possible to use a 20MHz clock source - I would like to address it in a separate PR so I'm not blocking @jessebraham any longer :D
Also, CTS and DTS need to be added.

### CODE REVIEW

File: esp-hal-procmacros/src/lib.rs [hunk 1]
```diff
@@ -705,28 +706,45 @@ pub fn entry(args: TokenStream, input: TokenStream) -> TokenStream {
...
-                    return parse::Error::new(arg.span(), "invalid argument to main")
-                        .to_compile_error()
-                        .into();
-                }
-                let pin = extract_pin(&t.ty);
-                if used_pins.contains(&pin) {
-                    return parse::Error::new(arg.span(), "duplicate pin")
-                        .to_compile_error()
-                        .into();
+                if get_simplename(&t.ty) == "GpioPin" {
```
Comment (bjoernQ) [inline esp-hal-procmacros/src/lib.rs:724]: maybe a `match` would look cleaner here

File: esp-hal-procmacros/src/lib.rs [hunk 2]
```diff
@@ -705,28 +706,45 @@ pub fn entry(args: TokenStream, input: TokenStream) -> TokenStream {
     let f = parse_macro_input!(input as ItemFn);
 
     let mut argument_types = Vec::new();
+    let mut create_peripheral = Vec::new();
 
     let mut used_pins: Vec<u8> = Vec::new();
+    let mut num: u32 = 0;
+
     for arg in &f.sig.inputs {
```
Comment (bjoernQ) [inline esp-hal-procmacros/src/lib.rs:714]: maybe we can use `.enumerate()` to get rid of the manual counting?

**Comment by bjoernQ** [comment]
Great we finally have this! Just some minor remarks from my side

**Review by bjoernQ — APPROVED** [review]
LGTM - Thank you!

**Review by MabezDev — APPROVED** [review]
LGTM, thank you!

**Pull request merged**